### PR TITLE
Log cleanup

### PR DESCRIPTION
**Pull request by megoth**
This PR includes migration to TS, unit tests, documentation, and examples.

### CODE REVIEW

**Comment by michielbdejong** [comment]
Great! Will review after the standup.

File: src/log.ts [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+ * - Warning: 2
+ * - Message: 4
+ * - Success: 8
+ * - Info: 16
+ * - Debug: 32
+ *
+ * You can also choose to set all by passing the number *64*.
+ */
+export function setLevel (level: number): void {
+  _level = TALL
```
Comment (michielbdejong) [inline src/log.ts]: What does this line do?

File: src/log.ts [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+ * sum of the levels you want to show:
+ *
+ * - Error: 1
+ * - Warning: 2
+ * - Message: 4
+ * - Success: 8
+ * - Info: 16
+ * - Debug: 32
+ *
+ * You can also choose to set all by passing the number *64*.
```
Comment (michielbdejong) [inline src/log.ts]: Wouldn't 63 make more sense instead of 64?

File: examples/log/index.html [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+    <ul>
+        <li>Error: 1</li>
+        <li>Warning: 2</li>
+        <li>Message: 4</li>
+        <li>Success: 8</li>
+        <li>Info: 16</li>
+        <li>Debug: 32</li>
+    </ul>
+    <p>
+        Setting level to 64 will show all messages.
```
Comment (michielbdejong) [inline examples/log/index.html]: 63, not 64. You could also clarify "It's a bitmask, so setting level to 63 (i.e. 1+2+4+8+16+32) will show all messages."

File: src/log.ts (hunk 2 above)
Comment (megoth) [reply src/log.ts]: Oh, yes, my bad, will fix

File: src/log.ts (hunk 1 above)
Comment (megoth) [reply src/log.ts]: Just making sure that the next line are shown, could be set to TDEBUG instead.

File: examples/log/index.html (hunk 3 above)
Comment (megoth) [reply examples/log/index.html]: Yeah, was wondering whether or not to mention the bitmask thing, as some might be confused by it. But I'll phrase it in such a way that is useful for those that understand and not necessary to understand for those who don't.

**Pull request merged**